application/models/Files.php

```diff
@@ -139,9 +139,26 @@ public static function addFile(Model_Users $user, Array $fileData, $thumbnail_da
 			return self::PROCESS_OK;
 		}
 		
-		public static function addFolder(Model_Users $user, $name = null, $description = null, Model_Files $parent = null)
+		public static function addFolder(Model_Users $user, $name = null, $description = null, Model_Files $parent = null, $tags = null)
 		{
-			$folder	=	new Model_Files($name, $description, 1, $user, $parent);
+			
+			//We manage the tags
+			$arrayTags	=	explode(' ', $tags);
+			$arrayTagsInstances = [];
+			
+			foreach($arrayTags as $tagName)
+			{
+				$tag = Model_Tags::getTag($tagName);
+				if(!$tag)
+				{
+					$newTag = new Model_Tags($tagName);
+					$tag = Model_Tags::add($newTag);
```

Comment:
Donc, j'ai apporté les modifs nécessaires et maintenant j'ai ça :

Warning: Creating default object from empty value in C:\wamp\www\ShyComics\application\vendors\EntityPHP\Entity.php on line 831
Call Stack
# Time Memory Function Location
1 0.0028 258424 {main}( ) ..\index.php:0
2 0.1420 1342592 Eliya\Request->exec( ) ..\index.php:39
3 0.1637 1503496 invokeArgs ( ) ..\Request.class.php:298
4 0.1637 1504104 Controller_spritecomics_gallery->post_index( ) ..\Request.class.php:298
5 0.1675 1621152 Model_Files::addFolder( ) ..\gallery.php:63
6 0.1782 1649200 EntityPHP\Entity::addMultiple( ) ..\Files.php:171

( ! ) Fatal error: Call to undefined method stdClass::prepareDataForSQL() in C:\wamp\www\ShyComics\application\vendors\EntityPHP\Entity.php on line 834
Call Stack
# Time Memory Function Location
1 0.0028 258424 {main}( ) ..\index.php:0
2 0.1420 1342592 Eliya\Request->exec( ) ..\index.php:39
3 0.1637 1503496 invokeArgs ( ) ..\Request.class.php:298
4 0.1637 1504104 Controller_spritecomics_gallery->post_index( ) ..\Request.class.php:298
5 0.1675 1621152 Model_Files::addFolder( ) ..\gallery.php:63
6 0.1782 1649200 EntityPHP\Entity::addMultiple( )